src/integer/mat_z/vector/is_vector.rs

```diff
@@ -0,0 +1,162 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module includes all functionality to check
+//! whether a matrix represents a vector, i.e. has only one row
+//! or one column.
+//! These methods should be used to ensure that vector functions
+//! can only be called on suitably formed vector/matrices.
+
+use super::super::MatZ;
+use crate::traits::{GetNumColumns, GetNumRows};
+
+impl MatZ {
+    /// Returns `true` if the provided [`MatZ`] has only one row,
+    /// i.e. is a row vector. Otherwise, returns `false`.
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer::MatZ;
+    /// use std::str::FromStr;
+    ///
+    /// let vec = MatZ::from_str("[[1,2,3]]").unwrap();
+    ///
+    /// assert!(vec.is_row_vector());
+    /// assert!(!vec.transpose().is_row_vector());
+    /// ```
+    pub fn is_row_vector(&self) -> bool {
+        self.get_num_rows() == 1
+    }
+
+    /// Returns `true` if the provided [`MatZ`] has only one column,
+    /// i.e. is a column vector. Otherwise, returns `false`.
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer::MatZ;
+    /// use std::str::FromStr;
+    ///
+    /// let vec = MatZ::from_str("[[1],[2],[3]]").unwrap();
+    ///
+    /// assert!(vec.is_column_vector());
+    /// assert!(!vec.transpose().is_column_vector());
+    /// ```
+    pub fn is_column_vector(&self) -> bool {
+        self.get_num_columns() == 1
+    }
+
+    /// Returns `true` if the provided [`MatZ`] has only one column or one row,
+    /// i.e. is a vector. Otherwise, returns `false`.
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer::MatZ;
+    /// use std::str::FromStr;
+    ///
+    /// let vec = MatZ::from_str("[[1],[2],[3]]").unwrap();
+    ///
+    /// assert!(vec.is_vector());
+    /// assert!(vec.transpose().is_vector());
+    /// ```
+    pub fn is_vector(&self) -> bool {
+        self.is_column_vector() || self.is_row_vector()
+    }
+
+    /// Returns `true` if the provided [`MatZ`] has only one entry,
+    /// i.e. is a 1x1 matrix. Otherwise, returns `false`.
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer::MatZ;
+    /// use std::str::FromStr;
+    ///
+    /// let vec = MatZ::from_str("[[1]]").unwrap();
+    ///
+    /// assert!(vec.has_single_entry());
+    /// ```
+    pub fn has_single_entry(&self) -> bool {
+        self.is_column_vector() && self.is_row_vector()
+    }
```

Comment:
I do not quite get the pupose of this function, but if yout think that we need it, feel free to keep it

Reply:
This function could become interesting, e.g. we want to automatically unpack single entry matrices for arithmetic operations. But yes, currently, we do not have any use cases.